### DISCUSSION TITLE
Invert resize transform for predicted masks

### DISCUSSION THREAD

**Discussion by sepidhk** [Project-MONAI/MONAI#6395] [Q&A] [answered]
Thank you for this excellent library. I have been using MONAI for both 2D and 3D medical image segmentation and both tasks have been successful. For a project, I performed a 2D segmentation and the original image size was (1164x873). In order to fit the images into the network, I had to use the "Resized" function in the transform to resize the images to (896x896), and I trained the model and predicted the test images successfully. Now, I need to resize the test images back to their original shapes for further processing steps. Is there an inverse transform function for 2D images that I can use to run predictions on the original images, similar to what we do for 3D volumes using "transforms.Invertd"? Alternatively, do you have any other solution to help me resize the images back to their original shapes? Note that I used the OpenCV "resize" function, but unfortunately, the prediction and input images do not overlap correctly. I would appreciate your help and time.

Here is my transform function for test data:

```
img_size = (896, 896)
test_transforms_encoded = Compose([
    LoadImaged(keys=["img", "label"], reader=PILReader),
    EnsureChannelFirstd(keys=["img", "label"]),
    ScaleIntensityd(keys=["img"], allow_missing_keys=True),
    Resized(keys=["img", "label"], spatial_size=img_size),
    Flipd(keys=["img", "label"], spatial_axis=0),
    Rotate90d(keys=["img", "label"], k=3),
    EnsureTyped(keys=["img", "label"]),
    AsDiscreted(keys=["label"], to_onehot=num_classes),
])
test_ds_encoded = monai.data.Dataset(data=test_files, transform=test_transforms_encoded)
test_loader_encoded = DataLoader(test_ds_encoded, batch_size=1, shuffle=False, num_workers=1)

```

Here is the prediction and saving the predicted masks:
```
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
normalize = ScaleIntensity()
post_trans = Compose([Activations(softmax=True)])
from tensorflow.keras.preprocessing.image import img_to_array, array_to_img, load_img

with torch.no_grad():
    for test_data in test_loader_encoded:
        test_inputs = test_data["img"].to(device)
        test_data["pred"] = sliding_window_inference(test_inputs, roi_size, sw_batch_size, model)
        test_outputs, test_labels = from_engine(["pred", "label"])(test_data)
        test_outputs = test_outputs.to(device)
        test_labels = test_labels.to(device)
        test_outputs = [post_trans(i) for i in decollate_batch(test_outputs)]
        test_labels = decollate_batch(test_labels)
        dice_score = dice_metric(y_pred=test_outputs, y=test_labels)
        iou_score = iou_metric(y_pred=test_outputs, y=test_labels)
        predicted = torch.argmax(test_data["pred"], dim=1, keepdim=True).detach().cpu()
        predicted_mask = np.moveaxis(np.squeeze(predicted, 0), 0, -1)
        filename = os.path.basename(test_data["img_meta_dict"]["filename_or_obj"][0])
        predicted_filename = os.path.join(out_dir, f"predicted_{filename}")
        print(f"{filename} ==> dice score: {dice_score.mean().item()}, iou metric: {iou_score.mean().item()}")
        predicted_mask_array = array_to_img(predicted_mask)
        predicted_mask_array.save(os.path.join(out_dir, filename))

```

**Comment by KumoLiu** [comment, marked as answer]
Hi @sepidhk, I think you could do the same thing just like what you do in 3d data. Most transforms don't have any distinction between 2d and 3d data. And for more details, you could refer to this tutorial which using a 2D dataset:
https://github.com/Project-MONAI/tutorials/blob/main/modules/inverse_transforms_and_test_time_augmentations.ipynb

Hope it can help you, thanks!
